File: Java/DrawingBook.java
public class DrawingBook {
    public static int pageCount(int n, int p) {
        int frontFlip = (int) Math.floor(p / 2);
        int backFlip = (int) Math.floor((n / 2) - frontFlip);

        int result = Math.min(frontFlip, backFlip);
        return result;
    }
    
    public static void main(String[] args) {
        // Test Cases
        System.out.println(pageCount(5, 3)); // expected output: 1
        System.out.println(pageCount(9, 5)); // expected output: 2
    }
}

File: JavaScript/DrawingBook.js
function pageCount(n, p) {
    var frontFlip = Math.floor(p / 2);
    var backFlip = Math.floor((n / 2) - frontFlip);

    let result = Math.min(frontFlip, backFlip);
    return result;

}

console.log(pageCount(5, 3))
console.log(pageCount(9, 5))